File: string-paradigm-dependency_2/src/main/java/com/jesperancinha/string/paradigm/performance2/StringWrapper.java
package com.jesperancinha.string.paradigm.performance2;

import java.io.UnsupportedEncodingException;
import java.util.Arrays;

public class StringWrapper {

	byte[] bytes;

	public StringWrapper(String string) {
		try {
			bytes = string.getBytes("ISO-8859-1");
		} catch (UnsupportedEncodingException e) {
			e.printStackTrace();
		}
	}

	public String toString() {
		String to;
		try {
			to = new String(bytes, "ISO-8859-1");
		} catch (UnsupportedEncodingException e) {
			e.printStackTrace();
			to = null;
		}
		return to;
	}

	public byte[] getBytes(int from, int to) {
		return Arrays.copyOfRange(bytes, from, to);
	}
}

File: string-paradigm_1/src/main/kotlin/org/jesperancinha/string/paradigm/performance1/StringWrapper.kt
package org.jesperancinha.string.paradigm.performance1

import java.io.UnsupportedEncodingException
import java.nio.charset.Charset
import java.util.*

class StringWrapper(string: String) {
    private val bytes: ByteArray by lazy { string.toByteArray(charset("ISO-8859-1")) }
    override fun toString() = String(bytes, Charset.forName("ISO-8859-1"))

    fun getBytes(from: Int, to: Int): ByteArray {
        return bytes.copyOfRange(from, to)
    }
}